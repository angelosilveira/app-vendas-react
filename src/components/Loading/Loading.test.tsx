import '@testing-library/jest-dom';
import { renderWithTheme } from 'src/utils/tests/renderWithTheme';
import { it, describe } from 'vitest';
import { Loading } from '.';

describe('<Loading />', () => {
  it('should render the loading', () => {
    const { container, getByTestId } = renderWithTheme(
      <Loading data-testid="loading" />
    );

    expect(getByTestId('loading')).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
