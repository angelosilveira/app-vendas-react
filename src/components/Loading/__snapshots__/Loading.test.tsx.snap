// Vitest Snapshot v1

exports[`<Loading /> > should render the loading 1`] = `
.c0 {
  display: inline-block;
  position: relative;
  width: 64px;
  height: 64px;
}

.c0 div {
  position: absolute;
  background: #6e17c8;
  opacity: 1;
  border-radius: 50%;
  -webkit-animation: cwzpQT 1.4s cubic-bezier(0,0.2,0.8,1) infinite;
  animation: cwzpQT 1.4s cubic-bezier(0,0.2,0.8,1) infinite;
}

.c0 div:nth-child(2) {
  -webkit-animation-delay: -0.7s;
  animation-delay: -0.7s;
}

<div
  class="c0"
  data-testid="loading"
>
  <div />
  <div />
</div>
`;
